docs/hugo/content/contributing/developer-setup.md

```diff
@@ -0,0 +1,174 @@
+3. Run the VS Code command (with `Ctrl-Shift-P`): `Dev Containers: Clone Repository in Container Volume...`
+
+   **Note**: in Windows, it is important to clone directly into a container instead of cloning first and then loading that with the `Remote Containers` extension, as the tooling performs a lot of file I/O, and if this is performed against a Windows folder mounted into the devcontainer then it is unusably slow.
+
+4. To complete the clone:
+    1. Select "`GitHub`".
+    2. Search the azure-service-operator repo.
+       * If you have direct access to the ASO repo, search for `Azure/azure-service-operator`
+       * Otherwise search for `<your-user-name>/azure-service-operator`
+    3. If asked, choose either of the following options about where to create the volume.
+    4. The window will reload and run the `Dockerfile` setup to build the devcontainer. The first time, this will take some minutes to complete as it installs all dependencies. Later runs will be faster due to caching.
+5. Once the repository is cloned in your local
+   * Make sure you have access to tags (by running `git tag --list 'v2*'`).
+   * Set up our submodule by running both `git submodule init` and `git submodule update`.
+   * See [Check your repo health](#troubleshooting-repo-health) for more information.
+6. To validate everything is working correctly,  open a terminal in VS Code and run `task -l`. This will show a list of all `task` commands. Running `task` by itself (or `task default`) will run quick local pre-checkin tests and validation.
+
+## Docker on Linux
+
+The same `Dockerfile` that the VS Code `devcontainer` extension uses can also be used outside of VS Code; it is stored in the root `.devcontainer` directory and can be used to create a development container with all the tooling preinstalled:
+
+```console
+$ docker build $(git rev-parse --show-toplevel)/.devcontainer -t asodev:latest
+… image will be created …
+
+$ # After that you can start a terminal in the development container with:
+$ docker run --env-file ~/work/envs.env --env HOSTROOT=$(git rev-parse --show-toplevel) -v $(git rev-parse --show-toplevel):/go/src -w /go/src -u $(id -u ${USER}):$(id -g ${USER}) --group-add $(stat -c '%g' /var/run/docker.sock) -v /var/run/docker.sock:/var/run/docker.sock --network=host  -it asodev:latest /bin/bash
+```
+
+Note: If you mount the source like this from a Windows folder, performance will be poor as file operations between the container and Windows are very slow.
+
+
+## CLI on Linux
+
+If you are using Linux, instead of using VS Code you can run the `dev.sh` script in the root of the repository. This will install all required tooling into the `hack/tools` directory and then start a new shell with the `PATH` updated to use it.
+
+## CLI on MacOS
+
+Development of ASO on MacOS is possible (one of our team does so), but things are less automated.
+
+You'll need to manually install the tools as listed by `.devcontainer/install-dependencies.sh`.
```

Comment:
If we don't have it already - Should we have a list of tools/dependencies somewhere? 

Just makes it easier for the users, so that they don't have to dig through the `.sh` files

Reply:
Good idea. I'll log a separate task to track doing this.